### PR TITLE
Added huffman_coding.c

### PR DESCRIPTION
**Pull request by YashovardhanSiramdas**
**Changes:**
Added Huffman Coding in C

### CODE REVIEW

File: code/greedy_algorithms/src/huffman_coding/huffman_coding.c [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        i = (i - 1) / 2;
+    }
+ 
+    minHeap->array[i] = minHeapNode;
+}
+
+void
+buildMinHeap(struct MinHeap* minHeap)
+{
+    int n = minHeap->size - 1;
```
Comment (arnavb) [inline code/greedy_algorithms/src/huffman_coding/huffman_coding.c]: I would get rid of this variable altogether, and use `minHeap->size - 2` in the loop.

File: code/greedy_algorithms/src/huffman_coding/huffman_coding.c [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+{
+    return !(root->left) && !(root->right);
+}
+
+struct MinHeap* 
+createAndBuildMinHeap(char data[], int freq[], int size)
+{
+    struct MinHeap* minHeap = createMinHeap(size);
+ 
+ 	int i;
```
Comment (arnavb) [inline code/greedy_algorithms/src/huffman_coding/huffman_coding.c]: Indentation.

**Comment by YashovardhanSiramdas** [comment]
@arnavb Thanks for the warm welcome !! Very glad to see that repository is flooded with new PR's !! :smile:

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**